jaeger-core/src/main/java/io/jaegertracing/JaegerSpan.java

```diff
@@ -206,6 +206,11 @@ public synchronized JaegerSpan setTag(String key, Number value) {
   }
 
   private JaegerSpan setTagAsObject(String key, Object value) {
+    if (finished) {
+      log.warn("Span has already been finished; cannot set tag.", new RuntimeException("Called from:"));
```

Comment:
Do you have a sample output for this log message? I'm not sure creating a new exception here (even if not throwing it) is what we want...

Reply:
Exception seems useful, but dual-finish doesn't use it currently, so if we want to have exception, I'd say it should be a tracer config switch to enable it.

If we are making this change for tags, it should also apply for all methods that write to the span, i.e. setOperation, log(), etc.

Reply:
I do agree that the exception seems useful, but I'd rather have two log entries then. One at warning level, with a message like "Cannot set tag, as the span is already closed. Set log level to debug for information about the caller". Then, the second would be in debug level, wrapped in `if (log.isDebugEnabled()) { logger.debug(...); }`, with the stack trace.

My concerns are:

1. Pollute the logs with stack traces
1. Confuse users into thinking these stack traces are actual exceptions
1. Wasted resources in collecting the stack trace for every single time the span is closed, as this information is useful only once per caller.